### PR TITLE
Add avahi based beacon for zeroconf announcement

### PR DESCRIPTION
**Pull request by adelcast**
### What does this PR do?

New beacon to announce services via avahi (zeroconf)
### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

Signed-off-by: Alejandro del Castillo alejandro.delcastillo@ni.com

### CODE REVIEW

File: salt/beacons/avahi_announce.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                       'beacon must be a dictionary')
+    elif not all(x in list(config.keys()) for x in ('servicetype', 'port', 'txt')):
+        return False, ('Configuration for avahi_announce beacon '
+                       'must contain servicetype, port and txt items')
+    return True, 'Valid beacon configuration'
+
+
+def beacon(config):
+    '''
+    Broadcast values via zeronconf
```
Comment (cachedout) [inline salt/beacons/avahi_announce.py]: zeroconf?

Comment (adelcast) [inline salt/beacons/avahi_announce.py]: indeed.....rebased and pushed an updated branch

**Comment by cachedout** [comment]
@adelcast Could you also please take care of this lint error? https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/3443/violations/file/salt/beacons/avahi_announce.py/

**Comment by adelcast** [comment]
sure thing!

**Comment by thatch45** [comment]
Thanks @adelcast ! This is a very cool beacon!

**Pull request merged**